src/pubsub/ua_pubsub_connection.c

```diff
@@ -200,6 +200,11 @@ UA_PubSubConnection_create(UA_Server *server,
     if(newConnectionsField->channel->openSubscriber) {
         newConnectionsField->channel->openSubscriber(newConnectionsField->channel);
     }
+
+    if (connectionConfig->enabled)
+    {
+        UA_Server_enablePubSubConnection(server, *connectionIdentifier);
```

Comment:
This could lead to a dead-lock.
Methods starting with `UA_Server_` are in the public API and take a lock to protect the server being accessed from parallel threads. Once you have the lock you must not take it a second time.

This method looks like you are already within the server-lock-context.
So don't call methods starting with UA_Server_ from here.
Ideally add UA_LOCK_ASSERT at the beginning of the method to make the assumptions explicit.